lib/manager/npm/post-update/index.ts

```diff
@@ -482,6 +486,9 @@ export async function getAdditionalFiles(
   for (const lockFile of dirs.yarnLockDirs) {
     const lockFileDir = path.dirname(lockFile);
     const fullLockFileDir = upath.join(config.localDir, lockFileDir);
+    const packageFile = packageFiles.npm.find(
+      (p) => path.dirname(p.packageFile) === lockFileDir
```

Comment:
Yarn's lock file directory may be different from the package file directory if Workspaces are used

Reply:
@rarkins As we're detecting the version of the package manager, the `compatibility` from the root package file (config) is enough.